src/main/kotlin/uk/gov/justice/digital/hmpps/externalusersapi/resource/GroupsController.kt

```diff
@@ -68,7 +72,61 @@ class GroupsController(
     val returnedGroup: Group = groupsService.getGroupDetail(group)
     return GroupDetails(returnedGroup)
   }
+
+  @PutMapping("/groups/child/{group}")
+  @PreAuthorize("hasRole('ROLE_MAINTAIN_OAUTH_USERS')")
+  @Operation(
+    summary = "Amend child group name.",
+    description = "Amend a Child Group Name"
+  )
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "OK"
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "Child Group not found.",
+        content = [
+          Content(
+            mediaType = "application/json",
+            schema = Schema(implementation = ErrorResponse::class)
+          )
+        ]
+      )
+    ]
+  )
+  fun amendChildGroupName(
+    @Parameter(description = "The group code of the child group.", required = true)
+    @PathVariable
+    group: String,
+    @Parameter(
+      description = "Details of the child group to be updated.",
+      required = true
+    ) @RequestBody
+    groupAmendment: GroupAmendment
+
+  ) {
+    groupsService.updateChildGroup(authenticationFacade.currentUsername, group, groupAmendment)
```

Comment:
We don't need to pass the username through here - the authenticationFacade can just be added to the constructor of the service.  I suppose it doesn't really matter, but this is how we did it for the RoleService.  Probably worth changing either RolesController/Service or GroupController/Service to be consistent.  It doesn't matter which.
Happy for this to be a separate PR

Reply:
Good point, Moved to service